optimum/exporters/onnx/base.py

```diff
@@ -378,6 +377,17 @@ def torch_to_onnx_input_map(self) -> Mapping[str, str]:
         """
         return {}
 
+    @property
+    def torch_to_onnx_output_map(self) -> Mapping[str, str]:
+        """
+        Dictionary of keys to update the ONNX output name for export. Override the function when
+        the output names and the exported ONNX output names need are different.
+
+        Returns:
+            `Mapping[str, str]`: A dictionary specifying the output name to exported ONNX output name map.
+        """
+        return {}
```

Comment:
I think you do not need that, you can use the `output_names_for_validation` method.

Reply:
It would be less efficient as the `output_names_for_validation` method uses `torch_to_onnx_output_map`

Reply:
How is it less efficient? The `output_names_for_validation` method exists and does what you need.

I am not in favor of adding yet a new method. `OnnxConfig` is already packed with methods, and in this specific case there is no real use of this one since you can use what alredy exists.

Reply:
The `output_names_for_validation` should use a dictionnary to map the torch and onnx model output names in order to be more general instead of returning a list with a fixed order of output names. Having a dictionnary to do the mapping  is needed for both the `ModelPatcher` and the validation step (`validate_model_outputs`), so I would rather remove the `output_names_for_validation` method and use the `torch_to_onnx_output_map` in `validate_model_outputs` directly

Reply:
I don't have a strong opinion on which one we should keep, but we should have only one of the two as they are doing mostly the same thing.